File: solution/1442. Count Triplets That Can Form Two Arrays of Equal XOR/1442_01.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        unordered_map<int, vector<int>> m;
        m[0].push_back(-1);
        int res = 0;
        for (int i = 0, prefix = 0; i < arr.size(); ++i) {
            prefix ^= arr[i];
            for (int idx : m[prefix]) {
                res += i - idx - 1;
            }
            m[prefix].push_back(i);
        }
        return res;
    }
};

File: solution/1442. Count Triplets That Can Form Two Arrays of Equal XOR/1442_01.py
class Solution:
    def countTriplets(self, arr: List[int]) -> int:
        m = defaultdict(list)
        m[0].append(-1)
        res = 0
        prefix = 0
        for i in range(len(arr)):
            prefix ^= arr[i]
            for idx in m[prefix]:
                res += i - idx - 1
            m[prefix].append(i)
        return res;
    
